### PR TITLE
Support Synth2Synth swaps

### PR DESCRIPTION
**Pull request by hammeiam**
Allow users to swap between synths, which is a different flow than our existing virtual swaps. They are shown a small notice that they will need to wait before transferring their tokens, but otherwise it is a normal swap. 

https://user-images.githubusercontent.com/7607886/128255921-f3fd59c6-9d55-4840-b2ed-91b45f639e97.mov

### CODE REVIEW

File: src/hooks/useApproveAndSwap.ts [hunk 1]
```diff
@@ -169,18 +181,27 @@ export function useApproveAndSwap(): (
...
+        console.debug("swap - synthToSynth", args)
+        swapTransaction = await baseSynthetixContract?.exchangeWithTracking(
+          ...args,
+        )
       } else {
         throw new Error("Invalid Swap Type, or contract not loaded")
       }
 
-      notifyHandler(swapTransaction.hash, "swap")
+      notifyHandler(swapTransaction?.hash || "", "swap")
```
Comment (alphastorm) [inline src/hooks/useApproveAndSwap.ts:204]: should we be calling `notifyHandler` with an empty tx hash?  wouldn't it be preferable to do something like:
```js
if (swapTransaction?.hash) {
  notifyHandler(swapTransaction?.hash, "swap")
}
```

**Comment by alphastorm** [comment]
To test:
- in saddle-contract on latest master
  - FUND_FORK_MAINNET=true npm run fork
- in saddle-frontend on david/pendingswaps-4-synth-x-synth
  - edit src/constants/index.ts to set IS_VIRTUAL_SWAP_ACTIVE = true
  - npm run start

**Comment by alphastorm** [comment]
transaction transaction!
Screen Shot 2021-08-23 at 3 27 37 PM

**Comment by penandlim** [comment]
Swapping between synths worked for me!
But when I tried to do swap from all SETH -> SBTC then back to SBTC - > SETH, I get this error in the console (which is expected). Would be great if we could explictly catch this or display a proper error message.

image

On a broader note: is there a way to show the rpc error message in the notify widget?

**Review by penandlim — APPROVED** [review]

File: src/constants/index.ts [hunk 2]
```diff
@@ -88,6 +88,20 @@ export class Token {
...
+export const SYNTHETIX_CONTRACT_ADDRESSES: { [chainId in ChainId]: string } = {
+  [ChainId.MAINNET]: "0xC011a73ee8576Fb46F5E1c5751cA3B9Fe0af2a6F", //"0x43AE8037179a5746D618DA077A38DdeEa9640cBa",
+  [ChainId.ROPSTEN]: "",
+  [ChainId.HARDHAT]: "",
+}
+
+export const SYNTHETIX_EXCHANGE_RATES_CONTRACT_ADDRESSES: {
+  [chainId in ChainId]: string
+} = {
+  [ChainId.MAINNET]: "0xd69b189020EF614796578AfE4d10378c5e7e1138",
```
[the remark this answers is not in the document]
Comment (hammeiam) [reply src/constants/index.ts:107]: Do you know if there's an automatic way of detecting when the address changes?

**Pull request merged**